custom-stats-logger/src/main/java/dev/gemfire/stat/logger/StatLogger.java

```diff
@@ -0,0 +1,151 @@
+  public void execute(FunctionContext functionContext) {
+    ClassLoader contextClassLoader = this.getClass().getClassLoader();
+    Properties properties = new Properties();
+    try {
+      final InputStream stream = contextClassLoader.getResourceAsStream("statLogger.properties");
+      properties.load(stream);
+    } catch (IOException e) {
+      logger.error("Could not load statLogger.properties", e);
+    }
+    String logFrequency = properties.getProperty("logFrequency");
+    String statistics = properties.getProperty("statistics");
+
+    cleanup(functionContext.getCache());
+
+    if (!StringUtils.isEmpty(statistics) && !StringUtils.isEmpty(logFrequency)) {
+      createTimerRegion(functionContext.getCache());
+
+      timerInterval = Long.parseLong(logFrequency);
+      createStatsList(functionContext.getCache().getDistributedSystem(), statistics);
+      ensureTimerRunning(functionContext.getCache());
+      functionContext.getResultSender().lastResult("Logging Metric count " + listOfStats.size() +
+              " with timer interval set to " + timerInterval + " ms");
+
+    } else if (StringUtils.isEmpty(statistics) && !StringUtils.isEmpty(logFrequency)) {
+      functionContext.getResultSender().lastResult("No stats to log. StatLogger will not start.");
+
+    } else if (!StringUtils.isEmpty(statistics) && StringUtils.isEmpty(logFrequency)) {
+      createTimerRegion(functionContext.getCache());
+
+      createStatsList(functionContext.getCache().getDistributedSystem(), statistics);
+      ensureTimerRunning(functionContext.getCache());
+      functionContext.getResultSender().lastResult("Logging Metric count " + listOfStats.size() +
+              " with timer interval set to default of " + timerInterval + " ms");
+
+    } else {
+      functionContext.getResultSender().lastResult("No properties set for StatLogger. StatLogger will not start.");
+
+    }
+  }
+
+  private void createStatsList(DistributedSystem distributedSystem, String statistics) {
+    List<String> statisticsList = Arrays.asList(statistics.split(",", 0));
+    for (String statistic : statisticsList) {
+      String[] statsNames = statistic.split("[.]", 0);
+      addStats(distributedSystem, statsNames[0], statsNames[1]);
+    }
+  }
+
+  private static void createTimerRegion(Cache cache) {
+    RegionFactory<Object, Object> regionFactory = cache.createRegionFactory(RegionShortcut.LOCAL);
+    regionFactory.create("Timer");
+    logger.info("Created Timer region");
+  }
+
+  private void cleanup(Cache cache) {
+    cancelTimer(cache);
+    listOfStats.clear();
+  }
+
+  private void cancelTimer(Cache cache) {
+    Region<Object, Object> timerRegion = cache.getRegion("Timer");
+    if (timerRegion != null) {
+      Object statsLoggerTimer = timerRegion.get("statsLoggerTimer");
+      if (statsLoggerTimer != null) {
+        ((Timer) statsLoggerTimer).cancel();
+        logger.info("Stopped old statsLoggerTimer");
+        timerRegion.destroyRegion();
```

Comment:
Rather than always destroying the region each time the function is executed, would it be better to only destroy it if the list of stats is empty? Simply removing the old timer entry and replacing it with the new one would have less overhead.

Reply:
I think I have tried that route and felt like it was better to go this way of destroying it would clean up itself everytime and not leave some stale unnecessary region behind.